File: discord/src/commands/guild/settings_command.rs
use entity::server;
use i18n::{
    translator::Translator,
    types::{Currency, Language},
};
use twilight_model::{
    channel::message::MessageFlags,
    http::interaction::{InteractionResponseData, InteractionResponseType},
};

use crate::{embeds, types::exports::InteractionResponse};

pub async fn settings_command(
    translator: &Translator,
    server: &Option<server::Model>,
    language: &Language,
    curreny: &Currency,
) -> Result<InteractionResponse, anyhow::Error> {
    return Ok(InteractionResponse {
        data: Some(InteractionResponseData {
            flags: Some(MessageFlags::EPHEMERAL),
            embeds: Some(vec![embeds::settings_embed::settings_embed(
                translator, server, language, curreny,
            )]),
            allowed_mentions: None,
            attachments: None,
            choices: None,
            components: None,
            content: None,
            tts: None,
            custom_id: None,
            title: None,
        }),
        kind: InteractionResponseType::ChannelMessageWithSource,
    });
}

File: backend/src/discord/commands/settingsCommand.ts
import { InteractionResponseType, MessageFlags } from "discord-api-types/v10";
import { settingsEmbed } from "../embeds/settings";
import { Command } from "./_commandType";

export const settingsCommand: Command = {
	name: "settings",
	needsGuild: true,
	needsManageGuild: true,
	handle: async (props) => {
		props.ctx.respondWith(200, {
			type: InteractionResponseType.ChannelMessageWithSource,
			data: {
				flags: MessageFlags.Ephemeral,
				embeds: [settingsEmbed(props.dbServer, props.language, props.currency)],
			},
		});
	},
};
